packages/create-webpack-app/src/generators/init/default.ts

```diff
@@ -0,0 +1,272 @@
+          }
+          return true;
+        },
+      },
+    ],
+    actions: function (answers: Answers) {
+      const actions: ActionType[] = [];
+
+      switch (answers.langType) {
+        case "ES6":
+          dependencies.push("babel-loader", "@babel/core", "@babel/preset-env");
+          break;
+        case "Typescript":
+          dependencies.push("typescript", "ts-loader");
+          break;
+      }
+
+      if (answers.devServer) {
+        dependencies.push("webpack-dev-server");
+      }
+
+      if (answers.htmlWebpackPlugin) {
+        dependencies.push("html-webpack-plugin", "html-loader");
+      }
+
+      if (answers.workboxWebpackPlugin) {
+        dependencies.push("workbox-webpack-plugin");
+      }
+
+      if (answers.isPostCSS) {
+        dependencies.push("postcss-loader", "postcss", "autoprefixer");
+      }
+
+      if (answers.extractPlugin !== "No") {
+        dependencies.push("mini-css-extract-plugin");
+      }
+
+      if (answers.cssType !== "none") {
+        dependencies.push("style-loader", "css-loader");
+        switch (answers.cssType) {
+          case "SASS":
+            dependencies.push("sass-loader", "sass");
+            break;
+          case "LESS":
+            dependencies.push("less-loader", "less");
+            break;
+          case "Stylus":
+            dependencies.push("stylus-loader", "stylus");
+            break;
+        }
+      }
+      if (answers.extractPlugin !== "No") {
+        dependencies.push("mini-css-extract-plugin");
+      }
+
+      const files = ["./index.html", "webpack.config.js", "package.json", "README.md"];
+
+      switch (answers.langType) {
+        case "Typescript":
+          answers.entryPoint = "./src/index.ts";
+          files.push("tsconfig.json", answers.entryPoint as string);
+          break;
+        case "ES6":
+          answers.entryPoint = "./src/index.js";
+          files.push("babel.config.json", answers.entryPoint as string);
+          break;
+        default:
+          answers.entryPoint = "./src/index.js";
+          files.push(answers.entryPoint as string);
+          break;
+      }
+      if (answers.isPostCSS) {
+        files.push("postcss.config.js");
+      }
+
+      for (const file of files) {
+        actions.push({
+          type: "add",
+          path: join(answers.projectPath, file),
+          templateFile: join(plop.getPlopfilePath(), "../templates/init/default", `${file}.tpl`),
+          transform: (content: string) => ejs.render(content, answers),
+          force: true,
+        });
+      }
+      actions.push({
+        type: "pkgInstall",
+        path: plop.renderString("{{projectPath}}/", answers),
+        // Custom function don't automatically render hbs template as path hence manual rendering
+        packages: dependencies,
+      });
+      return actions;
+
+      // answers.entryPoint = answers.langType === "Typescript" ? "./src/index.ts" : "./src/index.js";
```

Comment:
whys this commented out?

Reply:
This was a previous way of doing the file creation and template logic, which was more restrictive so I commented it out and didnt delete for reference.
I think Now there is no need for this will remove this in next commit.